litmus-portal/authentication/pkg/user/repository.go

```diff
@@ -86,8 +89,15 @@ func (r repository) CreateUser(user *entities.User) (*entities.User, error) {
 
 //UpdateUser updates user details in the database
 func (r repository) UpdateUser(user *entities.User) (*entities.User, error) {
+	if user.Password != "" {
+		hashedPassword, err := bcrypt.GenerateFromPassword([]byte(user.Password), utils.PasswordEncryptionCost)
+		if err != nil {
+			return nil, err
+		}
+		user.Password = string(hashedPassword)
+	}
 	data, _ := toDoc(user)
-	_, err := r.Collection.UpdateOne(context.Background(), bson.M{"_id": user.ID}, bson.M{"$set": data})
+	_, err := r.Collection.UpdateOne(context.Background(), bson.M{"username": user.UserName}, bson.M{"$set": data})
```

Comment:
Updating via username would be the right approach, but I think if we are migrating to updaing via username, we should get the username parsed from `JWT parsing` and not from the request body sent by frontend(client), as anyone can mimic a request with the username and update the details.

Reply:
Why do we need to use username instead of id @SarthakJain26 ?

Reply:
@gdsoumya  ID and username both are unique, I updated it to username because on initial login where user is asked to update his/her password, we are sending username from the frontend. 
It was not giving any error because ID and username both have `omitempty` tags, which I think should be removed. cc: @DarthBenro008

Reply:
I think the jwt token will have the id right? we are getting the jwt token in the backend or are we not sending the token in this call?

Reply:
Yes we are getting token. I think `JWT parsing` is good for both username as well as ID

Reply:
I think user id would be better then.

Reply:
>I think user id would be better then.
then frontend needs to change the params Right?

Reply:
No, the data is being taken from jwt right so it should always be there @SarthakJain26 ?

Reply:
> No, the data is being taken from jwt right so it should already be there @SarthakJain26 ?

Yes @gdsoumya